@alexbruy a couple of notes after testing this:

- I get an error if I try to edit an existing enum option:

```
TypeError: setAllowMultiple() takes 1 positional argument but 2 were given 
Traceback (most recent call last):
  File "/home/nyall/dev/build-QGIS/output/python/plugins/processing/modeler/ModelerGraphicItem.py", line 435, in mousePressEvent
    self.action()
  File "/home/nyall/dev/build-QGIS/output/python/plugins/processing/modeler/ModelerGraphicItem.py", line 210, in editElement
    param=self.model.parameterDefinition(self.element.parameterName()))
  File "/home/nyall/dev/build-QGIS/output/python/plugins/processing/modeler/ModelerParameterDefinitionDialog.py", line 83, in __init__
    self.setupUi()
  File "/home/nyall/dev/build-QGIS/output/python/plugins/processing/modeler/ModelerParameterDefinitionDialog.py", line 285, in setupUi
    self.widget.setAllowMultiple(bool(self.param.allowMultiple()))
TypeError: setAllowMultiple() takes 1 positional argument but 2 were given
```
- I can set multiple default checks even when "allow multiple selection" is not checked. 
- If I check "allow multiple selection" and then check one or more default options, I get an exception when running the alg:
```
Traceback (most recent call last):
  File "/home/nyall/dev/build-QGIS/output/python/plugins/processing/modeler/ModelerDialog.py", line 334, in runModel
    dlg = AlgorithmDialog(self.model)
  File "/home/nyall/dev/build-QGIS/output/python/plugins/processing/gui/AlgorithmDialog.py", line 70, in __init__
    self.setMainWidget(self.getParametersPanel(alg, self))
  File "/home/nyall/dev/build-QGIS/output/python/plugins/processing/gui/AlgorithmDialog.py", line 77, in getParametersPanel
    return ParametersPanel(parent, alg)
  File "/home/nyall/dev/build-QGIS/output/python/plugins/processing/gui/ParametersPanel.py", line 81, in __init__
    self.initWidgets()
  File "/home/nyall/dev/build-QGIS/output/python/plugins/processing/gui/ParametersPanel.py", line 110, in initWidgets
    wrapper = WidgetWrapperFactory.create_wrapper(param, self.parent)
  File "/home/nyall/dev/build-QGIS/output/python/plugins/processing/gui/wrappers.py", line 1670, in create_wrapper
    return WidgetWrapperFactory.create_wrapper_from_class(param, dialog, row, col)
  File "/home/nyall/dev/build-QGIS/output/python/plugins/processing/gui/wrappers.py", line 1734, in create_wrapper_from_class
    return wrapper(param, dialog, row, col)
  File "/home/nyall/dev/build-QGIS/output/python/plugins/processing/gui/wrappers.py", line 160, in __init__
    self.setValue(param.defaultValue())
  File "/home/nyall/dev/build-QGIS/output/python/plugins/processing/gui/wrappers.py", line 1001, in setValue
    self.widget.setSelectedItems(value)
  File "/home/nyall/dev/build-QGIS/output/python/plugins/processing/gui/MultipleInputPanel.py", line 64, in setSelectedItems
    self.tr('{0} elements selected').format(len(self.selectedoptions)))
TypeError: object of type 'int' has no len()
```
- I think the "clear" icon should be `console/iconClearConsole.svg` instead of the checkbox clear (and maybe a confirmation prompt should be added there?)

Matrix param:
- I also get an exception if I try to edit an existing matrix parameter
- I think the widget should default to one column - a zero column matrix isn't very useful :)
- If i enter a 2x2 matrix with values 1/2/3/4, then the last entered value is forgotten when clicking "OK", (unless you do some other step after entering it - like adding a new column)

But otherwise it's working very nicely!

--- a/python/plugins/processing/gui/matrixmodelerwidget.py
+++ b/python/plugins/processing/gui/matrixmodelerwidget.py
@@ -122,7 +122,7 @@ class MatrixModelerWidget(BASE, WIDGET):
         headers = []
         model = self.tblView.model()
         for i in range(model.columnCount()):
-            headers.append(model.headerData(i, Qt.Horizontal))
+            headers.append(str(model.headerData(i, Qt.Horizontal)))
 
         return headers
 
@@ -133,5 +133,5 @@ class MatrixModelerWidget(BASE, WIDGET):
     def fixedRows(self):
         return self.chkFixedRows.isChecked()
 
-    def setFixedRows(self):
-        self.chkFixedRows.setChecked(True)
+    def setFixedRows(self, fixedRows):
+        self.chkFixedRows.setChecked(fixedRows)

--- a/python/plugins/processing/gui/wrappers.py
+++ b/python/plugins/processing/gui/wrappers.py
@@ -64,6 +64,7 @@ from qgis.core import (
     QgsProcessingParameterFeatureSource,
     QgsProcessingParameterMapLayer,
     QgsProcessingParameterBand,
+    QgsProcessingParameterMatrix,
     QgsProcessingParameterDistance,
     QgsProcessingFeatureSourceDefinition,
     QgsProcessingOutputRasterLayer,
@@ -556,19 +557,26 @@ class FileWidgetWrapper(WidgetWrapper):
 class FixedTableWidgetWrapper(WidgetWrapper):
 
     def createWidget(self):
-        return FixedTablePanel(self.param)
+        if self.dialogType in (DIALOG_STANDARD, DIALOG_BATCH):
+            return FixedTablePanel(self.param)
+        else:
+            self.combobox = QComboBox()
+            values = self.dialog.getAvailableValuesOfType(QgsProcessingParameterMatrix)
+            for v in values:
+                self.combobox.addItem(self.dialog.resolveValueDescription(v), v)
+            return self.combobox
 
     def setValue(self, value):
-        self.widget.setValue(value)
+        if self.dialogType in (DIALOG_STANDARD, DIALOG_BATCH):
+            self.widget.setValue(value)
+        else:
+            self.setComboValue(value, combobox=self.combobox)
 
     def value(self):
-        if self.dialogType == DIALOG_MODELER:
-            table = self.widget.table
-            if not bool(table) and not self.param.flags() & QgsProcessingParameterDefinition.FlagOptional:
-                raise InvalidParameterValue()
-            return ParameterFixedTable.tableToString(table)
-        else:
+        if self.dialogType in (DIALOG_STANDARD, DIALOG_BATCH):
             return self.widget.table
+        else:
+            return self.comboValue(combobox=self.combobox)
 
 
 class MultipleLayerWidgetWrapper(WidgetWrapper):
@@ -959,40 +967,53 @@ class RasterWidgetWrapper(MapLayerWidgetWrapper):
 class EnumWidgetWrapper(WidgetWrapper):
 
     def createWidget(self, useCheckBoxes=False, columns=1):
-        self._useCheckBoxes = useCheckBoxes
-        if self._useCheckBoxes and not self.dialogType == DIALOG_BATCH:
-            return CheckboxesPanel(options=self.param.options(),
-                                   multiple=self.param.allowMultiple(),
-                                   columns=columns)
-        if self.param.allowMultiple():
-            return MultipleInputPanel(options=self.param.options())
+        if self.dialogType in (DIALOG_STANDARD, DIALOG_BATCH):
+            self._useCheckBoxes = useCheckBoxes
+            if self._useCheckBoxes and not self.dialogType == DIALOG_BATCH:
+                return CheckboxesPanel(options=self.param.options(),
+                                       multiple=self.param.allowMultiple(),
+                                       columns=columns)
+            if self.param.allowMultiple():
+                return MultipleInputPanel(options=self.param.options())
+            else:
+                widget = QComboBox()
+                for i, option in enumerate(self.param.options()):
+                    widget.addItem(option, i)
+                if self.param.defaultValue():
+                    widget.setCurrentIndex(widget.findData(self.param.defaultValue()))
+                return widget
         else:
-            widget = QComboBox()
-            for i, option in enumerate(self.param.options()):
-                widget.addItem(option, i)
-            if self.param.defaultValue():
-                widget.setCurrentIndex(widget.findData(self.param.defaultValue()))
-            return widget
+            self.combobox = QComboBox()
+            values = self.dialog.getAvailableValuesOfType(QgsProcessingParameterEnum)
+            for v in values:
+                self.combobox.addItem(self.dialog.resolveValueDescription(v), v)
+            return self.combobox
 
     def setValue(self, value):
         if value is None or value == NULL:
             return
 
-        if self._useCheckBoxes and not self.dialogType == DIALOG_BATCH:
-            self.widget.setValue(value)
-            return
-        if self.param.allowMultiple():
-            self.widget.setSelectedItems(value)
+        if self.dialogType in (DIALOG_STANDARD, DIALOG_BATCH):
+            if self._useCheckBoxes and not self.dialogType == DIALOG_BATCH:
+                self.widget.setValue(value)
+                return
+            if self.param.allowMultiple():
+                self.widget.setSelectedItems(value)
+            else:
+                self.widget.setCurrentIndex(self.widget.findData(value))
         else:
-            self.widget.setCurrentIndex(self.widget.findData(value))
+            self.setComboValue(value, combobox=self.combobox)
 
     def value(self):
-        if self._useCheckBoxes and not self.dialogType == DIALOG_BATCH:
-            return self.widget.value()
-        if self.param.allowMultiple():
-            return self.widget.selectedoptions
+        if self.dialogType in (DIALOG_STANDARD, DIALOG_BATCH):
+            if self._useCheckBoxes and not self.dialogType == DIALOG_BATCH:
+                return self.widget.value()
+            if self.param.allowMultiple():
+                return self.widget.selectedoptions
+            else:
+                return self.widget.currentData()
         else:
-            return self.widget.currentData()
+            return self.comboValue(combobox=self.combo)
 
 
 class FeatureSourceWidgetWrapper(WidgetWrapper):

--- a/python/plugins/processing/modeler/ModelerParameterDefinitionDialog.py
+++ b/python/plugins/processing/modeler/ModelerParameterDefinitionDialog.py
@@ -419,9 +419,12 @@ class ModelerParameterDefinitionDialog(QDialog):
         elif (self.paramType == parameters.PARAMETER_CRS or
               isinstance(self.param, QgsProcessingParameterCrs)):
             self.param = QgsProcessingParameterCrs(name, description, self.selector.crs().authid())
-        if (self.paramType == parameters.PARAMETER_ENUM or
+        elif (self.paramType == parameters.PARAMETER_ENUM or
                 isinstance(self.param, QgsProcessingParameterEnum)):
             self.param = QgsProcessingParameterEnum(name, description, self.widget.options(), self.widget.allowMultiple(), self.widget.defaultOption())
+        elif (self.paramType == parameters.PARAMETER_MATRIX or
+                isinstance(self.param, QgsProcessingParameterMatrix)):
+            self.param = QgsProcessingParameterMatrix(name, description, hasFixedNumberRows=self.widget.fixedRows(), headers=self.widget.headers(), defaultValue=self.widget.value())
         else:
             if self.paramType:
                 typeId = self.paramType

--- a/python/plugins/processing/ui/matrixmodelerwidgetbase.ui
+++ b/python/plugins/processing/ui/matrixmodelerwidgetbase.ui
@@ -28,6 +28,9 @@
    </property>
    <item row="0" column="0" rowspan="8">
     <widget class="QTableView" name="tblView">
+     <property name="selectionBehavior">
+      <enum>QAbstractItemView::SelectRows</enum>
+     </property>
      <attribute name="horizontalHeaderStretchLastSection">
       <bool>true</bool>
      </attribute>

--- a/src/core/processing/qgsprocessingregistry.cpp
+++ b/src/core/processing/qgsprocessingregistry.cpp
@@ -30,6 +30,7 @@ QgsProcessingRegistry::QgsProcessingRegistry( QObject *parent SIP_TRANSFERTHIS )
   addParameterType( new QgsProcessingParameterTypeCrs() );
   addParameterType( new QgsProcessingParameterTypeRange() );
   addParameterType( new QgsProcessingParameterTypePoint() );
+  addParameterType( new QgsProcessingParameterTypeEnum() );
   addParameterType( new QgsProcessingParameterTypeExtent() );
   addParameterType( new QgsProcessingParameterTypeMatrix() );
   addParameterType( new QgsProcessingParameterTypeFile() ) ;

